### ISSUE TITLE
Update New Supportal to differentiate between information that has been intentionally redacted from view and buggy behaviour

### ISSUE THREAD

**Issue by VictoriaExpensify** [Expensify/App#51449]
**Problem:**
We have a basic version of New Expensify Supportal for viewing user accounts, we need to add new features and improve reliability so that it can be used to support customers and provide demos to prospective new ones.

At the moment, if someone tries to access information in Supportal that has been redacted, all they see is ongoing spinners. This is the same thing we see when there is a bug or issue that is preventing us from viewing information. This is making it very difficult to determine what is working as expected in Supportal, and what is a bug.

Example:
This is a DM and should be redacted (which it is):
![image](https://github.com/user-attachments/assets/9e89978b-cbda-4f38-80d1-71a5d321bd96)

This is an expense that I should be able to view, so it appears to be a bug:
![image](https://github.com/user-attachments/assets/e53f7495-5f9d-4e01-b165-68322f876b2d)

**Solution:**
When Supportalled into a customers account, if we try to view information that is purposely redacted, an error should load stating that this information is not available when support logged in. This was discussed here - https://expensify.slack.com/archives/C07NZ8B1VTQ/p1728500635598069

**What is redacted?**
- Certain chat_types are currently redacted, the PR that removes these chats is [here](https://github.com/Expensify/Auth/pull/10366)
- The redacted chat logic will be updated soon [with this issue](https://github.com/Expensify/App/issues/51221)
- We also redact virtual card details from the supportal view

**Comment by mallenexpensify** [comment]
Ha, a bug AND new feature. I removed ya @sakluger , I'll snag this as I'm chatting with @stephanieelliott about how manage these.  Adding `Internal` and `Daily`

**Comment by mallenexpensify** [comment]
It's Internal, hot pick and daily.  Need to try to find someone to help and/or wait for it to get picked up.

**Comment by marcaaron** [comment]
>if someone tries to access information in Supportal that has been redacted, all they see is ongoing spinners.

I think we can definitely show a generic error in the case where supportal tries to _anything_ that it shouldn't (we do that in OldDot so a similar solution could be done for NewDot). But would love to understand the root of why access is even attempted in the first place.

**Comment by marcaaron** [comment]
Looks like we have logic [here](https://github.com/Expensify/Auth/blob/4a0789d4ed4b794be13f838d9c56852c0e639b60/auth/command/OpenReport.cpp#L209C1-L212C1) to return "nothing" but not to throw any specific error message. Here's one proposal...

Supportal can't access something we can return a specific jsonCode and message like `423 Unable to complete action as support user`. Then we can check for the 423 in NewDot and show an alert like:

![Screenshot 2024-11-01 at 12 07 48 PM](https://github.com/user-attachments/assets/d99027a0-7755-4a3e-982e-6db8ada8d487)

Does that work for now? cc @thienlnam @danieldoglas

**Comment by thienlnam** [comment]
Yeah that's a good idea, we can just have all those flows throw the same error and eventually have an external issue making it pretty if we want

**Comment by marcaaron** [comment]
Oh yeah also I think it's `411 Insufficient privileges`... let's see...

**Comment by marcaaron** [comment]
Oh wait no, that is a pretty generic one... I'll come up with something 😄

**Comment by marcaaron** [comment]
Hmm so I'm not sure if I understand the premise of this issue after looking into it a bit...

If we can't access the report as a supportal user an error is thrown and this screen is displayed:

![Screenshot 2024-11-01 at 1 03 00 PM](https://github.com/user-attachments/assets/75977414-1e6f-43d0-bc71-3ea8c39fa030)

Can someone provide reproduction steps for the exact issue we are trying to solve?

**Comment by mallenexpensify** [comment]
> Can someone provide reproduction steps for the exact issue we are trying to solve?

@VictoriaExpensify , can you , since you're way more familiar with supportal and the OP? thx

**Comment by VictoriaExpensify** [comment]
Oof, you know what, Supportal isn't showing the long-loading chats anymore (like I've shown in my initial screenshots). This is kind of a perfect example of why seeing if things have been purposefully redacted/blocked is important. But also makes it hard to reproduce 😅 

Regardless, I was going to suggest we put this on hold while this issue is worked:

- [CRITICAL] Update New Supportal to include chat details #51221

That is going to indirectly resolve a lot of these issues because it will be clear when messages and chats are redacted. Once that improvement is made it will be easier to pin-point any screens where it's not clear if info has been redacted and we can decide if this issue needs to be reprioritised.

**Comment by mallenexpensify** [comment]
On Hold, leaving weekly

**Comment by VictoriaExpensify** [comment]
Oh this has predominantly been resolved through Marc's work in this issue - https://github.com/Expensify/App/issues/51221#issuecomment-2533330892

Closing this
